drivers/spi/spi_context.h

```diff
@@ -57,6 +59,24 @@ struct spi_context {
 #define SPI_CONTEXT_INIT_SYNC(_data, _ctx_name)				\
 	._ctx_name.sync = Z_SEM_INITIALIZER(_data._ctx_name.sync, 0, 1)
 
+#define SPI_CONTEXT_CS_GPIO_SPEC_ELEM(_node_id, _prop, _idx)				\
+	{										\
+		.gpio_dev = DEVICE_DT_GET(DT_GPIO_CTLR_BY_IDX(_node_id, _prop, _idx)),	\
+		.gpio_pin = DT_GPIO_PIN_BY_IDX(_node_id, _prop, _idx),			\
+		.gpio_dt_flags = DT_GPIO_FLAGS_BY_IDX(_node_id, _prop, _idx),		\
+	},
```

Comment:
this is duplicating `GPIO_DT_SPEC_GET_BY_IDX`, but with different field names, why not modify SPI to use gpio_dt_spec?

Reply:
spi_cs_control struct does not use gpio_dt_spec so I've decided to not touch this and use it as it's and convert that later when it will be done. Take a look at the above conversation.

Reply:
I think it's better to _modernize_ SPI other than adding duplicate code on top of it.

Reply:
Sure, I just didn't want to duplicate the work that @JordanYates has done here https://github.com/csiro-wsn/zephyr/commits/210801_spi_cs

Reply:
I'll change that when https://github.com/zephyrproject-rtos/zephyr/pull/39116 will be merged.

Reply:
Fixed.